docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,172 @@
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveconfig, you can either provide the new configurations (aka `clusterDeploymentRelatedMetrics`) or the old way of hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only.
+
+#### Failure modes
+
+These situations will cause a panic().
+- clusterDeploymentRelatedMetrics.metrics[].name doesn't exist as a metric
+- clusterDeploymentRelatedMetrics.metrics[].name is a metric without duration but minimumDuration was specified.
+- a metric is defined across multiple groupings and the customizations defined contradict. For ex, following hiveConfig will cause panic because hive will be confused whether to parse `minimumDuration` for `hive_foo_seconds` to be 10m or 1h.
+  ```
+  HiveConfig:
+    spec:
+      metricsConfig:
+        clusterDeploymentRelatedMetrics:
+        - metrics:
+          - hive_foo_seconds
+          minimumDuration: 10m
+        - metrics:
+          - hive_foo_seconds
+          - hive_foo_gauge
+          minimumDuration: 1h
+  ```
+- additionalClusterDeploymentLabels conflict with existing fixed label for that metric
+- if deprecated methods are used along with the new clusterDeploymentRelatedMetrics. You can either choose the _old_ way or the _new_ way.
+
+
+### Risks and Mitigations
+The biggest risk is the sheer number of metrics that are going to be affected. Roughly 29 metrics that we report right now can be classified as related to clusterDeployment.
+
+Moreover, 8 of those metrics do not have `minimumDuration` support right now will be able to have that customization, and we would need ideally a non-hardcoded way  of classifying them as supported or not.
+
+About 19 metrics related to cluster deployment which do not currently have support for `additionalClusterDeploymentLabels` will now have that support. This might create an avenue for worsening the cardinality, however, that would be the responsibility of Hive admin.
+
+It can get confusing for consumers to know the fixed labels each metric has. Update the [hive_metrics](https://github.com/openshift/hive/blob/master/docs/hive_metrics.md) doc to list out all labels and keep it up-to-date.
```

Comment:
I know I mentioned this before as a pie-in-the-sky possibility: can we say we'll at least _try_ to investigate having the docs generated from code so they're guaranteed to stay up to date? Even if it ends up being a card on the backlog forever...

Reply:
Nice, thanks.

Reply:
Can you please link that card here? ("We may in the future attempt to auto-generate these docs -- see [HIVE-2413](https://issues.redhat.com/browse/HIVE-2413)")

Reply:
done